airflow/config_templates/config.yml

```diff
@@ -2331,6 +2331,14 @@ scheduler:
       type: integer
       example: ~
       default: "8974"
+    num_stuck_in_queued_retries:
```

Comment:
I am late to join the party but I don't follow why we need this. We have [AIRFLOW__SCHEDULER__STANDALONE_DAG_PROCESSOR](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#task-queued-timeout) which introduced in Airflow 2.6

Reply:
task queued timeout consumes the retries on the task as set by the dag author. This is a problem when the task is not idempotent and therefore deliberately has retries set to 0. But the particular failure of a task to even get to running should not actually consume the task's retries that are for the running of the task. So we still need the timeout setting, but we also need a separate counter for attempts to get from queued to running, whereas retries are attempts to get from running to success.

Reply:
so yeah, these are very much different things.
do we need `num_stuck_in_queued_retries` to be configurable? not necessarily.

Reply:
> so yeah, these are very much different things.
do we need num_stuck_in_queued_retries to be configurable? not necessarily.

That is where I was going. This sounds more like airflow internal mechanism. I think we better not expose this and possibly cause confusion how settings are interacting and what they are about

Reply:
Alright.  I have made this an undocumented param.  Hopefully this pleases everyone.  If a sophisticated user (like an airflow-as-a-service provider) wanted to customize this, they can.  But for average user, it is invisible.

And making it configurable-albeit-hidden is a bonus for test readability because we can control exactly how many time it should retry, and this is important to control for testing.  Resolving convo.  PTAL